includes/discovery/discovery-protocols.inc.php

```diff
@@ -221,21 +221,31 @@ function ($v) {
                 }
             }
         }
-
-        $dot1d_array = snmpwalk_group($device, 'dot1dBasePortIfIndex', 'BRIDGE-MIB');
-        $lldp_ports = snmpwalk_group($device, 'lldpLocPortId', 'LLDP-MIB');
+        if (($device['os'] == 'aos7')) {
+            $lldp_local = snmpwalk_cache_oid($device, 'lldpLocPortEntry', [], 'LLDP-MIB');
+            $lldp_ports = snmpwalk_group($device, 'lldpLocPortId', 'LLDP-MIB');
+        } else {
+            $dot1d_array = snmpwalk_group($device, 'dot1dBasePortIfIndex', 'BRIDGE-MIB');
+            $lldp_ports = snmpwalk_group($device, 'lldpLocPortId', 'LLDP-MIB');
+        }
     }
 
     foreach ($lldp_array as $key => $lldp_if_array) {
         foreach ($lldp_if_array as $entry_key => $lldp_instance) {
-            if (is_numeric($dot1d_array[$entry_key]['dot1dBasePortIfIndex'])) {
+            if (($device['os'] == 'aos7')) {
+                $ifName = $lldp_local[$entry_key]['lldpLocPortDesc'];
```

Comment:
instead of this $ifName and the SELECT you do below, wouldn't it be possible to tweak ` $local_port_ifName = $lldp_ports[hexdec($lldp_ports_num[$key]['mtxrNeighborInterfaceID'])]['mtxrInterfaceStatsName'];` so that all the remaining code can be kept unchanged ?

Reply:
The idea would be to avoid any dbFetchRow (we are trying to get rid of them, and at least not to add any new one). So if you can get the $ifIndex right, and $local_port_id as well, then get_port_by_id will do it's magic. At the end, code is easier to read and understand and maintain :)

Reply:
So now i got rid of `dbFetchRow` and i changed only `$local_port_id`
Does this look better now from your point of view?

Reply:
yup :)